include/slate/BaseMatrix.hh

```diff
@@ -1250,7 +1250,14 @@ template <typename scalar_t>
 void BaseMatrix<scalar_t>::gridinfo(
     GridOrder* order, int* nprow, int* npcol, int* myrow, int* mycol ) const
 {
-    if (nprow_ > 0) {
+    int mpi_size;
+    MPI_Comm_size(mpiComm(), &mpi_size);
+    if (mpi_size == 1) {
+        *order = GridOrder::Col;
+        *nprow = *npcol = 1;
+        *myrow = *mycol = 0;
+    }
+    else if (nprow_ > 0) {
```

Comment:
May I suggest reordering this a bit, to handle the case when someone creates a row-major 1x1 grid in a more expected fashion. Currently gridinfo would return a col-major 1x1 grid, instead of the expected row-major grid.
```cpp
    if (nprow_ > 0) {
        ... existing code ...
    }
    else {
        int mpi_size;
        slate_mpi_call(
            MPI_Comm_size( mpiComm(), &mpi_size ) );
        if (mpi_size == 1) {
            *order = GridOrder::Col;
            *nprow = *npcol = 1;
            *myrow = *mycol = 0;
        }
        else {
            *order = GridOrder::Unknown;
            *nprow = *npcol = *myrow = *mycol = -1;
        }
    }
```

Reply:
Yes sure,